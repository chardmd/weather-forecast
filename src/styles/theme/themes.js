const mongoDBTheme = {
  primary: '#4DB33D',
  text: '#589636',
  textSecondary: '#3F3E42',
  background: '##E8E7D5',
  backgroundVariant: '#E8E7D5',
  border: '#3F3E42',
  borderLight: '#C1BEBC',
};

export const themes = {
  mongoDBTheme: mongoDBTheme,
};
