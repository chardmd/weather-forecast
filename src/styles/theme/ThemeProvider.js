import * as React from 'react';
import { ThemeProvider as OriginalThemeProvider } from 'styled-components';
import { themes } from './themes';

export const ThemeProvider = props => {
  return (
    <OriginalThemeProvider theme={themes.mongoDBTheme}>
      {props.children}
    </OriginalThemeProvider>
  );
};
