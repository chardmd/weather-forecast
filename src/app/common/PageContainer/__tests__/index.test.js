import * as React from 'react';
import { render } from '@testing-library/react';

import PageContainer from '../index';
import { themes } from '../../../../styles/theme/themes';
import { ThemeProvider } from 'styled-components';

const renderWithTheme = (props, theme) =>
  render(
    <ThemeProvider theme={theme || themes.mongoInspiredTheme}>
      <PageContainer {...props} />
    </ThemeProvider>,
  );

describe('<PageContainer />', () => {
  it('should match snapshot', () => {
    const component = renderWithTheme();
    expect(component.container.firstChild).toMatchSnapshot();
  });
});
