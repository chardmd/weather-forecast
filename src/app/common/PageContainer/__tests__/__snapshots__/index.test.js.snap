// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PageContainer /> should match snapshot 1`] = `
.c0 {
  min-width: 250px;
  max-width: 960px;
  margin: 0 auto;
  padding: 0 1.5rem;
  box-sizing: content-box;
}

<div
  class="c0"
/>
`;
